models/darknet/darknet.hpp

```diff
@@ -0,0 +1,307 @@
+ * 
+ * For more information, kindly refer to the following paper.
+ * 
+ * Paper for DarkNet-19.
+ *
+ * @code
+ * @article{Redmon2016,
+ *  author = {Joseph Redmon, Ali Farhadi},
+ *  title = {YOLO9000 : Better, Faster, Stronger},
+ *  year = {2016},
+ *  url = {https://pjreddie.com/media/files/papers/YOLO9000.pdf}
+ * }
+ * @endcode
+ * 
+ * Paper for DarkNet-53.
+ * 
+ * @code
+ * @article{Redmon2016,
+ *  author = {Joseph Redmon, Ali Farhadi},
+ *  title = {YOLOv3 :  An Incremental Improvement},
+ *  year = {2019},
+ *  url = {https://pjreddie.com/media/files/papers/YOLOv3.pdf}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MODELS_DARKNET_HPP
+#define MODELS_DARKNET_HPP
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+#include <mlpack/methods/ann/layer/layer_types.hpp>
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+#include <mlpack/methods/ann/loss_functions/cross_entropy_error.hpp>
+#include <mlpack/methods/ann/init_rules/he_init.hpp>
+#include <mlpack/methods/ann/init_rules/glorot_init.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */{
+
+/**
+ * Definition of a Darknet CNN.
+ * 
+ * @tparam OutputLayerType The output layer type used to evaluate the network.
+ * @tparam InitializationRuleType Rule used to initialize the weight matrix.
+ * @tparam DaknetVer Version of DarkNet.
+ */
+template<
+  typename OutputLayerType = CrossEntropyError<>,
+  typename InitializationRuleType = RandomInitialization,
+  size_t DarkNetVersion = 19
+>
+class DarkNet
+{
+ public:
+  //! Create the DarkNet model.
+  DarkNet();
+
+  /**
+   * DarkNet constructor intializes input shape and number of classes.
+   *
+   * @param inputChannels Number of input channels of the input image.
+   * @param inputWidth Width of the input image.
+   * @param inputHeight Height of the input image.
+   * @param numClasses Optional number of classes to classify images into,
+   *                   only to be specified if includeTop is  true.
+   * @param weights One of 'none', 'cifar10'(pre-training on CIFAR10) or path to weights.
+   * @param includeTop Must be set to true if weights are set.
+   */
+  DarkNet(const size_t inputChannel,
+          const size_t inputWidth,
+          const size_t inputHeight,
+          const size_t numClasses = 1000,
+          const std::string& weights = "none",
+          const bool includeTop = true);
+
+  /**
+   * DarkNet constructor intializes input shape and number of classes.
+   *
+   * @param inputShape A three-valued tuple indicating input shape.
+   *                   First value is number of Channels (Channels-First).
```

Comment:
I think it shouldn't be capitalized
```suggestion
   *                   First value is number of channels (Channels-First).
```